src/libmain.c

```diff
@@ -1231,6 +1230,9 @@ gint main_lib(gint argc, gchar **argv)
 #endif
 
 	gtk_main();
+#ifdef G_OS_WIN32
+	win32_free_argv_made_in_utf8(argc, argv);
```

Comment:
Alternatively the conversion and freeing could be done in `main()` itself (from *main.c*), so `main_lib()` simply receives a nice argument list directly:

```C
int main(int argc, char **argv)
{
	int ret;
#ifdef G_OS_WIN32
	/* on Windows, get the Unicode argulment list and convert it to UTF-8 into argv[] */
	LPWSTR *szarglist = CommandLineToArgvW(GetCommandLineW(), &argc);

	argv = g_new(char *, argc + 1);
	for (int i = 0; i < argc; i++)
		argv[i] = g_utf16_to_utf8((gunichar2 *)szarglist[i], -1, NULL, NULL, NULL);
	argv[argc] = NULL;
	LocalFree(szarglist);
#endif

	ret = main_lib(argc, argv);

#ifdef G_OS_WIN32
	g_strfreev(argv);
#endif

	return ret;
}
```